samples/net/secure_mqtt_sensor_actuator/src/tls_config/config-tls-app.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2024 Analog Devices, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __CONFIG_TLS_APP_H__
+#define __CONFIG_TLS_APP_H__
+
+/* Application-specific Mbed TLS configuration.
+ * The required config will depend on the certificates being used.
+ * This file is included at the end of Zephyr's generic configuration.
+ */
+
+#define MBEDTLS_PK_HAVE_ECC_KEYS
```

Comment:
Are all those configs really needed? First, we should aim to configure mbed TLS with Kconfig only (pretty much of those can be configured in prj.conf), but what's more important, the `mqtt_publisher` sample seems to be able to establish TLS connection with `test.mosquitto.org` running pretty much on defaults:
https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/net/mqtt_publisher/overlay-tls.conf#L5
The mosquito certificate seems to rely on RSA encryption (enabled by default) not EC.

Reply:
You're right - not all needed. These were the configs I needed for the particular certs I was using with a local Mosquitto instance. I opted to leave them there to serve as a reference as to how you can configure MbedTLS in your app.
I realise is is probably best to disable any unneeded features so I removed this extra config file and Kconfig symbols.